src/common/entities/resources.entity.ts

```diff
@@ -0,0 +1,30 @@
+import {
+  Entity,
+  PrimaryKey,
+  Property,
+  ManyToOne,
+  EntityRepositoryType,
+  Rel,
+} from "@mikro-orm/core";
+
+import { ResourcesRepository } from "@/resources/resources.repository";
+
+import { Classroom } from "./classrooms.entity";
+import { CustomBaseEntity } from "./custom-base.entity";
+
+@Entity({ tableName: "resources", repository: () => ResourcesRepository })
+export class Resources extends CustomBaseEntity {
+  [EntityRepositoryType]?: ResourcesRepository;
+
+  @PrimaryKey()
+  id!: number;
+
+  @Property({ fieldName: "title" })
+  title!: string;
+
+  @Property({ fieldName: "file_url" })
+  fileUrl!: string;
+
+  @ManyToOne(() => Classroom)
+  classroom!: Rel<Classroom>;
+}
```

Comment:
Don't you need other fields as well to accomodate different resource types?

Reply:
I kept resource table different, this table will contain the shared pdf or class notes. I will create another table for assignments, their I will have deadline and descriptions column

Reply:
If you're planning to keep it separate with no relationships please discuss the decisions in the standup with Ragib Bhai first